File: chor-extraction/src/main/java/ast/sp/nodes/Network.java
package ast.sp.nodes;

import ast.sp.interfaces.SPNode;

import java.util.Collections;
import java.util.List;

public class Network implements SPNode{
    private final List<ProcessBehaviour> network;
    private boolean projectable;

    public Network(List<ProcessBehaviour> network) {
        this.network = network;
        projectable = true;
    }

    public List<ProcessBehaviour> getNetwork() {
        return network;
    }

    public boolean isProjectable() {
        return projectable;
    }

    public void setProjectable(boolean projectable) {
        this.projectable = projectable;
    }

    public String toString(){
        StringBuilder builder = new StringBuilder();
        Collections.sort(network, (pb1, pb2) -> pb1.getProcess().compareTo(pb2.getProcess()));

        for (ProcessBehaviour entry : network) {
            builder.append(entry.toString()).append(" | ");
        }

        if (builder.length()>=3){
            builder.delete(builder.length()-3, builder.length());
        }

        return builder.toString();
    }
}

File: chor-extraction/src/main/java/ast/sp/nodes/Network.kt
package ast.sp.nodes

import ast.sp.interfaces.SPNode
import java.util.*

data class Network(val network: TreeMap<String, ProcessBehaviour>) : SPNode {
    override fun toString(): String {
        val builder = StringBuilder()
        network.forEach { t, u -> builder.append(t).append(u.toString()).append(" | ")}
        if (builder.length >= 3) { builder.delete(builder.length - 3, builder.length) }
        return builder.toString()
    }

    fun copy(): Network{
        val temp = network.clone()
        return Network(temp as TreeMap<String, ProcessBehaviour>)
    }

    fun equals(n: Network): Boolean{
        for(p in network) {
            val process = p.key
            val np = n.network.get(process)
            if (np!=null){
                if (!p.value.main.equals(np.main)) return false
                for (pr in p.value.procedures){
                    val procedure = pr.key
                    val npr = np.procedures.get(procedure)
                    if ((npr==null) || !pr.value.equals(npr)) return false
                }
            } else return false
        }
        return true
    }
}


